### DISCUSSION TITLE
doc: Documentation issue in 'services/device_mgmt/smp_groups/smp_group_8' FS MGMT File Upload Request

### DISCUSSION THREAD

**Discussion by sgfeniex** [zephyrproject-rtos/zephyr#82078] [General]
**Describe the bug**

The Zephyr SMP `File upload` section is unclear with respect to the required encoding of the `data` payload information.

The current documentation lists file upload request payload as:

```
{
    (str)"off"      : (uint)
    (str)"data"     : (str)
    (str)"name"     : (str)
    (str,opt)"len"  : (uint)
}
```

where the file download response is:

```
{
    (str)"off"      : (uint)
    (str)"data"     : (byte str)
    (str,opt)"len"  : (uint)
}
```

In an upload request, `data` **must** by a `byte str` by way of:
`ZCBOR_MAP_DECODE_KEY_DECODER("data", zcbor_bstr_decode, &file_data),`
see: https://github.com/zephyrproject-rtos/zephyr/blob/b88d0bf8c7ee5fa308e9c6c8642d3efc6cb7064e/subsys/mgmt/mcumgr/grp/fs_mgmt/src/fs_mgmt.c#L367

"CBOR data of request" as well as the description of the "data" field need to be updated to reflect that only a `ZCBOR_MAJOR_TYPE_BSTR` is accepted.

**Environment**

* Page: `services/device_mgmt/smp_groups/smp_group_8`
* Version: main
* SHA-1: 5a5fd7c96fbdf1a80191b39296625a6174cfdd42
* https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/services/device_mgmt/smp_groups/smp_group_8.rst

**Comment by fabiobaltieri** [comment]
@nordicjm can you take a look please?

**Comment by JPHutchins** [comment]
Interesting, the smp package is definitely only supporting bytes: https://jphutchins.github.io/smp/latest/file_management/#smp.file_management.FileUploadRequest
